penaltymodel_core/penaltymodel/core/classes/penaltymodel.py

```diff
@@ -159,6 +159,8 @@ def __init__(self, graph, decision_variables, feasible_configurations, vartype,
 
         if not isinstance(model, BinaryQuadraticModel):
             raise TypeError("expected 'model' to be a BinaryQuadraticModel")
+        if set(model.variables) != set(graph.nodes):
```

Comment:
It's a bit expensive to cast to set for both, could maybe do `set(model.variables).symmetric_difference(graph.nodes)`? Though these are small enough it doesn't really matter.

Reply:
Cool! Thanks for pointing out this function.